vulcanai/models/metrics.py

```diff
@@ -615,14 +620,16 @@ def _run_test_multi(network, data_loader, plot=False, save_path=None,
             results : dict
 
         """
-        num_classes = network._num_classes
+        num_classes = network.num_classes
 
         if num_classes > 2:
             average = "macro"
         elif num_classes == 2:
             logger.warning("Will report scores only for pos_label, which is \
                            set to {}".format(pos_label))
             average = "binary"
+        else:
+            raise ValueError("Incorrect number of classes for run_test_multi")
```

Comment:
I think you should print what was received and what should be expected to make this error more actionable - or at least use that info to further debug

Reply:
this would never really happen unless someone overwrote the code. but I added more detail as well.